one_fm/hiring/doctype/onboard_employee/onboard_employee.js

```diff
@@ -145,6 +154,18 @@ frappe.ui.form.on('Onboard Employee', {
 	}
 });
 
+
+
+var hide_fields = function(frm){
+	let value = null
+	frm.doc.employment_type == "Service Provider" ? value = 1 : value = 0
+	frm.set_df_property('visa_and_residency_section','hidden',value)
+	frm.set_df_property('g2g_and_residency_section','hidden',value)
+	frm.set_df_property('uniform_measurements_section','hidden',value)
+	frm.set_df_property('work_contract_section','hidden',value)
+	frm.set_df_property('tools_for_work_section','hidden',value)
```

Comment:
You can hide the fields by depends_on from  doctype definition, since the employment_type field is in the doctype.